### PR TITLE
feat: validator for ferry trips + bikes allowance

### LINKED ISSUE #1499: Update validator with new Best Practice (bike_allowed present for ferry trips)
Add a rule with a WARNING severity level that checks is `bike_allowed` is present for ferry trips, after the addition in https://github.com/MobilityData/GTFS_Schedule_Best-Practices/pull/55.

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**

Add a rule with a WARNING severity level that checks is bike_allowed is present for ferry trips, after the addition in https://github.com/MobilityData/GTFS_Schedule_Best-Practices/pull/55.
Closes #1499 

**Expected behavior:** 

missing_bike_allowance warning notice is added to the report if ferry route types don't specify the bikes_allowed value. Valid values are 1 (bikes allowed) and 2 (bikes not allowed). Any value different from 1 or 2 will result in the warning being present in the report.

image

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by isabelle-dr** [comment]
Thank you for working on this! 
This looks great 👍.

**Comment by isabelle-dr** [comment]
@davidgamez, does this PR have any dependencies with the automatic generation of documentation?

**Review by isabelle-dr — APPROVED** [review]

File: RULES.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ Each Notice is associated with a severity: `INFO`, `WARNING`, `ERROR`.
 | [`unusable_trip`](#unusable_trip)                                                             | Trips must have more than one stop to be usable.                                                                                                              |
 | [`unused_shape`](#unused_shape)                                                               | Shape is not used in GTFS file `trips.txt`.                                                                                                                   |
 | [`unused_trip`](#unused_trip)                                                                 | Trip is not be used in `stop_times.txt`                                                                                                                       |
+| [`missing_bike_allowance_info`](#missing_bike_allowance_info)                                 | `bikes_allowed` value should be specified for ferry trips                                                                                                     |
```
Comment (jcpitre) [inline RULES.md:142]: Don't you think that having _info at the end could lead to confusion since this is actually a warning?

Comment (cka-y) [reply RULES.md:142]: Agreed. I meant by this name "the bike allowance information is missing". What name would you suggest?

Comment (davidgamez) [reply RULES.md:142]: suggestion: what about if we remove the suffix and leave it as: `missing_bike_allowance`

** Please merge master to this branch as per #1517; We are not updating RULES.md anymore. The name changes need to be done only in the Notice Java class name.

**Comment by jcpitre** [comment]
@cka-y I think it would be worthwhile to verify that the failing acceptance tests are consistant with the change in the code and were expected.
In such a situation I would say it's good practice to explain the test failures in a PR comment.

**Comment by cka-y** [comment]
Since we introduced a new warning notice, some acceptance tests are failing. 
After examination of the evaluation report, I can confirm that all differences are expected and related to the newly added warning notice.

**Review by jcpitre — APPROVED** [review]

**Pull request merged**